core/domain/rights_manager.py

```diff
@@ -862,7 +862,10 @@ def _deassign_role(committer, removed_user_id, activity_id, activity_type):
                 committer_id, removed_user_id, activity_id))
         raise Exception(
             'UnauthorizedUserException: Could not deassign role.')
-
+    if committer_id == removed_user_id:
```

Comment:
Move this check to the controller layer!

Reply:
Done!